### PR TITLE
Use 'Evaluee/recipient' in student submission UI #8176

### LINKED ISSUE #8176: Use 'Evaluee/recipient' in student submission UI
At some point we changed the feedback submission form to use the term _Evaluee_ instead of _Recipient_. However, some parts of the UI still uses the old term, this sentence for example:
> Instructors in this course can see your response, the name of the recipient, and your name.

We need to use the new term everywhere.

### PR DESCRIPTION
**Pull request by sukanta-27**
Fixes #8176 

**Outline of Solution**

Moved the `Evaluee/Recipient` from `response.tag` to `questionWithResponses.tag` to avoid repeatative use of the word `Evaluee`, and modified the names of recipients to show in bold font.

In addition to the UI changes shown in the screenshots in https://github.com/TEAMMATES/teammates/issues/8176#issuecomment-353736462, I also made the some changes in the UI of `instructorHelpSessions.jsp` to keep a consistent look all over the UI [Screenshot below].

**UI before:** 
screenshot 275

**UI after:**
screenshot 280

**In addition:**
I think we should change the name of the issue and PR to match the current scope of the PR, as we are not anymore replacing all `recipient` to `evaluee` everywhere in the UI, but rather are modifying the Student Submission forms primarily. (https://github.com/TEAMMATES/teammates/issues/8176#issuecomment-352323106)

### CODE REVIEW

**Comment by sukanta-27** [comment]
Realised that moving the code for showing `Evaluee/Recipient`  from `response.tag` to `questionWithResponses.tag`, created the problem where `Evaluee/Recipient` is being used in places it is not supposed to. (because the `questionWithResponses.tag` is being used by every kind of questions). Screenshots below.

screenshot 285
<br>
screenshot 283
<br>
screenshot 281

Looking for a way to solve this problem.
**In the meantime:** If I can get any **suggestions** or **hints** on how I can solve/approach this, will be very helpful as I am still a beginner at JSP and JSTL. (Though I know that finding how to solve is, is part of my expected work :sweat_smile: )

**Comment by sukanta-27** [comment]
The Current submission forms look like : 

**Instructor Feedback Submission Edit Page:**
screenshot 289

**Student Feedback Submission Edit Page:**

screenshot 290<br>

screenshot 291<br>
screenshot 292<br><br>

**Comment by sukanta-27** [comment]
@damithc Ready for review.

**Comment by damithc** [comment]
@wkurniawan07 for your review please.

File: src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag [hunk 1]
```diff
@@ -54,6 +56,20 @@
         </ul>
       </c:if>
 
+      <c:if test="${not isRecipientNameHidden}">
+        <div class="col-sm-12 form-inline mobile-align-left">
+          <label for="input"
+              style="text-indent: 24px">
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag]: This looks short enough to be put in the previous line.

File: src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag [hunk 2]
```diff
@@ -54,6 +56,20 @@
...
+              <span data-toggle="tooltip"
+                  data-placement="top"
+                  title="<%= Const.Tooltips.EVALUEE_DESCRIPTION %>">
+                  Evaluee/Recipient
+              </span>
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag]: This `<span>` block should be pushed to the left by 2 spaces. Additionally, the `Evaluee/Recipient` text.

File: src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/response.tag [hunk 3]
```diff
@@ -26,29 +26,24 @@
...
-    <label for="input">
-      <span data-toggle="tooltip"
-        data-placement="top"
-        title="<%= Const.Tooltips.EVALUEE_DESCRIPTION %>">
-        Evaluee
-      </span>
+    <label>
+      <select class="participantSelect middlealign<c:if test="${not response.existingResponse}"> newResponse</c:if> form-control"
+          name="<%= Const.ParamsNames.FEEDBACK_RESPONSE_RECIPIENT %>-${questionWithResponses.question.qnIndx}-${response.responseIndx}"
+          ${isNumResponsesMax ? 'style="display:none;max-width:125px"' : 'style="width:275px;max-width:275px"'}
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/response.tag]: You can extract out `style=""` here and only put the conditional inside, i.e. `style="${...}"`.

File: src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag [hunk 4]
```diff
@@ -7,6 +7,8 @@
 <%@ attribute name="isShowRealQuestionNumber" type="java.lang.Boolean" required="true" %>
 <%@ attribute name="isSessionOpenForSubmission" type="java.lang.Boolean" required="true" %>
 
+<c:set var="isRecipientNameHidden" value="${questionWithResponses.question.recipientNameHidden}"/>
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag]: Why is this the conditional used to determine whether the `Evaluee/Recipient` is displayed or not? What is the logic here?

Comment (sukanta-27) [reply src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag]: @wkurniawan07 
Without this conditional, `Evaluee/Recipient` was being displayed in Questions where there was no recipient name, like places like `Question 7` shown in the screenshot here : https://github.com/TEAMMATES/teammates/pull/8265#issuecomment-354286157, and other similar questions, and the UI did not look that much appealing for that reason.

File: src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag (hunk 2 above)
Comment (sukanta-27) [reply src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag]: Okay

File: src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag (hunk 1 above)
Comment (sukanta-27) [reply src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag]: Okay

File: src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/response.tag (hunk 3 above)
Comment (sukanta-27) [reply src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/response.tag]: Okay.

File: src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag (hunk 4 above)
Comment (wkurniawan07) [reply src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag]: That is not the issue. My question is why the condition to be used to determine whether `Evaluee/Recipient` is displayed is `questionWithResponses.question.recipientNameHidden`. How is that condition representative of whether or not to display `Evaluee/Recipient`?

Comment (sukanta-27) [reply src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag]: There was a similar test previously used in `response.tag` file, 
```jsp
${isRecipientNameHidden ?  'style="display:none"' : 'style="text-align:right"'}>
```
and I noticed that the questions which should have `Evaluee/Recipient` have recipient names like this one : 
screenshot 295

And the questions where the `Evaluee/Recipient` does not look appealing or appropriate does not have recipient names like this one : 
screenshot 281

So, I used `recipientNameHidden` variable to test if the recipient name is hidden in that question or not, and based on that I displayed the word `Evaluee/Recipient`.

File: src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag [hunk 5]
```diff
@@ -54,6 +56,17 @@
         </ul>
       </c:if>
 
+      <c:if test="${not isRecipientNameHidden}">
+        <div class="col-sm-12 form-inline mobile-align-left">
+          <label for="input" style="text-indent: 24px">
+            <span data-toggle="tooltip" data-placement="top" title="<%= Const.Tooltips.EVALUEE_DESCRIPTION %>">
+                Evaluee/Recipient
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag]: Move this to the left by 2 spaces.

Comment (sukanta-27) [reply src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag]: Done.

**Comment by damithc** [comment]
@sukanta-27 are the help pages updated to match this UI change?

**Comment by sukanta-27** [comment]
@damithc Sir,  I only encountered the word `Evaluee` in the Instructor help page under the `session` section and I have updated the file  `src/main/webapp/partials/instructorHelpSessions.jsp` accordingly.  I have added the screenshots of the UI changes in the instructorHelp page in this comment : https://github.com/TEAMMATES/teammates/pull/8265#issue-284903396

**Pull request merged**